### PR TITLE
Fix mingw builds

### PR DESCRIPTION
**Pull request by janisozaur**

### CODE REVIEW

**Comment by IntelOrca** [comment]
Closes #34

File: src/discord_register_win.cpp [hunk 1]
```diff
@@ -1,3 +1,9 @@
+#ifdef __MINGW32__
+// 0x0600 == vista
+#define WINVER 0x0600
```
Comment (crmarsh) [inline src/discord_register_win.cpp]: Any note on why this number? Also, seems cleaner to have this defined in build?

File: src/CMakeLists.txt [hunk 2]
```diff
@@ -44,6 +45,8 @@ if(WIN32)
         /wd4946 # reinterpret_cast used between related classes
         /wd5027 # move assignment operator was implicitly defined as deleted
     )
+    endif(MSVC)
+    target_link_libraries(discord-rpc PRIVATE psapi)
```
Comment (crmarsh) [inline src/CMakeLists.txt]: Yeah, this is better here either way.

File: src/discord_register_win.cpp (hunk 1 above)
Comment (janisozaur) [reply src/discord_register_win.cpp]: You are using Vista API: [RegSetKeyValue](https://msdn.microsoft.com/en-us/library/windows/desktop/ms724921(v=vs.85).aspx)

Comment (janisozaur) [reply src/discord_register_win.cpp]: As long as it is defined before any of the system headers get included, it's fine. In our project we put it in `common.h`. But if you prefer, I can move it to CMake.

Comment (crmarsh) [reply src/discord_register_win.cpp]: I know, sorry I was unclear, I meant, if you are gonna put some non-obvious stuff in code, it's nice to add a comment about it.

Comment (janisozaur) [reply src/discord_register_win.cpp]: I've added an explanation.

**Comment by crmarsh** [comment]
lgtm 👍

**Pull request merged**